File: 26-RemoveDuplicatesFromSortedArray/RemoveDuplicatesFromSortedArray.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int left = 1;
        if (nums.length<1){
            return 0;
        }
        for(int right=left; right<nums.length; right++){
            if (nums[right]!=nums[right-1]){
                nums[left]=nums[right];
                left++;
            }
        }
        return left;
    }
}

public class RemoveDuplicatesFromSortedArray {
    public static void main(String[] args) {
        Solution s = new Solution();
        int [] nums = {0,0,1,1,1,2,2,3,3,4};
        int uniquenumbers = s.removeDuplicates(nums);
        System.out.println(uniquenumbers);
    }
}

File: Leetcode/26-RemoveDuplicatesFromSortedArray/RemoveDuplicatesFromSortedArray.py
import typing
from typing import List
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        slow_pointer= 1
        if len(nums)<1 :
            return 0
        for fast_pointer in range(slow_pointer,len(nums)) :
            if nums[fast_pointer] != nums[fast_pointer-1] :
                nums[slow_pointer] = nums[fast_pointer]
                slow_pointer = slow_pointer +1
        return slow_pointer
s = Solution()
print(s.removeDuplicates([0,0,1,1,1,2,2,3,3,4]))